libethereum/BlockChain.cpp

```diff
@@ -84,17 +84,23 @@ std::ostream& dev::eth::operator<<(std::ostream& _out, BlockChain const& _bc)
 
 ldb::Slice dev::eth::toSlice(h256 const& _h, unsigned _sub)
 {
-#if ALL_COMPILERS_ARE_CPP11_COMPLIANT
-	static thread_local h256 h = _h ^ sha3(h256(u256(_sub)));
-	return ldb::Slice((char const*)&h, 32);
-#else
 	static boost::thread_specific_ptr<FixedHash<33>> t_h;
 	if (!t_h.get())
 		t_h.reset(new FixedHash<33>);
 	*t_h = FixedHash<33>(_h);
 	(*t_h)[32] = (uint8_t)_sub;
 	return (ldb::Slice)t_h->ref();//(char const*)t_h.get(), 32);
-#endif
+}
+
+ldb::Slice dev::eth::toSlice(uint64_t _n, unsigned _sub)
```

Comment:
Additional method is required for generic queryExtras method to work for both types. Casing to uint64_t in general case is wrong since both block hashes and numbers are stored in the database.